plugins/policy/cache/cache_impl.go

```diff
@@ -107,8 +107,13 @@ func (pc *PolicyCache) LookupPodsByNSLabelSelector(policyNamespace string,
 	mlPods := pc.getPodsByNSLabelSelector(policyNamespace, matchLabels)
 	mePods := pc.getMatchExpressionPods(policyNamespace, matchExpressions)
 
-	return utils.UnstringPodID(utils.Intersect(mlPods, mePods))
-
+	if len(matchLabels) > 0 && len(matchExpressions) > 0 {
+		return utils.UnstringPodID(utils.Intersect(mlPods, mePods))
+	}
+	if len(matchLabels) > 0 {
+		return utils.UnstringPodID(mlPods)
+	}
+	return utils.UnstringPodID(mePods)
```

Comment:
A little nit:  the name of the function should be utils.UnstringPodIDs (with and 's' at the end to indicate plural)